src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/EnumTypeModelBinderProvider.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// A <see cref="IModelBinderProvider"/> for types deriving from <see cref="System.Enum"/>.
+    /// </summary>
+    public class EnumTypeModelBinderProvider : IModelBinderProvider
+    {
+        private readonly MvcOptions _mvcOptions;
+
+        public EnumTypeModelBinderProvider(MvcOptions mvcOptions)
+        {
+            _mvcOptions = mvcOptions;
+        }
+
+        /// <inheritdoc />
+        public IModelBinder GetBinder(ModelBinderProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (context.Metadata.UnderlyingOrModelType.IsEnum)
```

Comment:
Since this is done infrequently, it's not a big deal. However `context.Metadata.IsEnum` avoids reflection when the information is already cached.